### PR TITLE
[cli] Add React/Relay component generator

### PR DESCRIPTION
**Pull request by alloy**
Supersedes #3.

The things missing are:

- [x] the `g` alias for the `generate` command
- [ ] ~~inferring the path at which to create the component from the component name~~
- [x] or if a path is given write the component to that
- [x] add test file template

### CODE REVIEW

**Comment by alloy** [comment]
Going to punt on the opinionated paths for now, so this is finally ready to review and ship!

File: packages/cli/src/__tests__/generator.test.ts [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+    compiledGenerator = compileGenerator()
+    expect.hasAssertions()
+  })
+
+  afterAll(() => {
+    fs.unlinkSync(compiledGenerator)
+  })
+
+  it("generates the component and test at the specified location", async () => {
+    ;({ sourceFile, testFile } = await generate(
```
Comment (orta) [inline packages/cli/src/__tests__/generator.test.ts]: `;({ sourceFile, testFile }` so weird - I get what it's doing but that's a funny way for (prettier) to do it, I guess it's the 100% safe from bad ASI

Comment (alloy) [reply packages/cli/src/__tests__/generator.test.ts]: It's how each old grumpy man's test should begin: 🤨

Comment (orta) [reply packages/cli/src/__tests__/generator.test.ts]: hah - yep

screen shot 2019-01-02 at 12 50 39 pm

**Comment by orta** [comment]
OK, cool, yep - this looks good, will get it deployed etc now

**Pull request merged**